src/otx/core/model/visual_prompting.py

```diff
@@ -966,47 +918,38 @@ def forward(  # type: ignore[override]
     def _customize_inputs(  # type: ignore[override]
         self,
         entity: ZeroShotVisualPromptingBatchDataEntity,  # type: ignore[override]
-    ) -> tuple[list[np.ndarray], list[dict[str, Any]], list[dict[int, list[Any]]]]:
+    ) -> tuple[list[np.ndarray], list[dict[str, Any]]]:
         """Customize OTX input batch data entity."""
         images: list[np.ndarray] = []
-        metas: list[dict[str, Any]] = []
-        processed_prompts: list[list[dict[str, Any]]] = []
-        for image, prompts, labels, imgs_info in zip(
+        processed_prompts: list[dict[str, Any]] = []
+
+        for image, prompts, labels in zip(
             entity.images,
             entity.prompts,
             entity.labels,
-            entity.imgs_info,
         ):
             # preprocess image encoder inputs
             numpy_image = image.cpu().numpy().transpose(1, 2, 0)
-            processed_image, meta = self.model["image_encoder"].preprocess(numpy_image)
-            images.append(processed_image)
-            metas.append(meta)
+            images.append(numpy_image)
+
             if self.training:
-                points: list[np.ndarray] = []
-                bboxes: list[np.ndarray] = []
-                _labels: dict[str, list[int]] = defaultdict(list)
+                points: list[Prompt] = []
+                bboxes: list[Prompt] = []
                 for prompt, label in zip(prompts, labels):  # type: ignore[arg-type]
                     if isinstance(prompt, tv_tensors.BoundingBoxes):
-                        bboxes.append(prompt.cpu().numpy())
-                        _labels["bboxes"].append(label.cpu().numpy())
+                        bboxes.append(Prompt(prompt.cpu().numpy(), label.cpu().numpy()))
                     elif isinstance(prompt, Points):
-                        points.append(prompt.cpu().numpy())
-                        _labels["points"].append(label.cpu().numpy())
+                        points.append(Prompt(prompt.cpu().numpy(), label.cpu().numpy()))
```

Comment:
Could you leave TODO comment about polygon support?

Reply:
Done